python/pyfory/serializer.py

```diff
@@ -849,6 +849,39 @@ def read(self, buffer):
             raise ValueError(f"Invalid reduce data format: {reduce_data[0]}")
 
 
+class FunctionSerializer(CrossLanguageCompatibleSerializer):
+    """Serializer for function objects using cloudpickle."""
+
+    def __init__(self, fory, cls):
+        super().__init__(fory, cls)
+        try:
+            import cloudpickle
+
+            self.cloudpickle = cloudpickle
+        except ImportError:
+            raise ImportError("cloudpickle is required for function serialization")
+
+    def xwrite(self, buffer, value):
+        # Use cloudpickle to serialize the function
+        pickled_data = self.cloudpickle.dumps(value)
```

Comment:
Could we support function serialization natively in fory. The function isntance may capture some closures objects, if we use cloudpickle/pickle to serializethe function, such objects will escape from fory object graph serialization. If the object is referenced outside of the function, it will be serialized by fory too, which will create two objects at deserialization. But this is wrong since we have only one reference at seriaization.

And another thing is that is also escape from fory class registration security checks

In the long wrong, we need to remove pickle/cloudpickle dependencies, to make fory as a drop-in replacement serialization framework.

Reply:
I replaced them with `fory.(de)serialize_ref` and marshal.